compiler/rustc_middle/src/ty/util.rs

```diff
@@ -124,7 +124,22 @@ impl IntTypeExt for IntegerType {
 impl<'tcx> TyCtxt<'tcx> {
     /// Creates a hash of the type `Ty` which will be the same no matter what crate
     /// context it's calculated within. This is used by the `type_id` intrinsic.
-    pub fn type_id_hash(self, ty: Ty<'tcx>) -> u64 {
+    pub fn type_id_hash_64(self, ty: Ty<'tcx>) -> u64 {
+        // We want the type_id be independent of the types free regions, so we
+        // erase them. The erase_regions() call will also anonymize bound
+        // regions, which is desirable too.
+        let ty = self.erase_regions(ty);
+
+        self.with_stable_hashing_context(|mut hcx| {
+            let mut hasher = StableHasher::new();
+            hcx.while_hashing_spans(false, |hcx| ty.hash_stable(hcx, &mut hasher));
+            hasher.finish()
+        })
+    }
+
+    /// Creates a hash of the type `Ty` which will be the same no matter what crate
+    /// context it's calculated within. This is used by the `type_id128` intrinsic.
+    pub fn type_id_hash_128(self, ty: Ty<'tcx>) -> u128 {
```

Comment:
YMMV: maybe make this generic, returning `<W: StableHasherResult>`, to avoid needing to copy-paste the implementation?

Reply:
I realized that I'm being dumb and don't need both `type_id` and `type_id128`, so this is a moot point.